comp/core/workloadmeta/collectors/internal/ecsfargate/v4parser_test.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+
+// Package ecsfargate implements the ECS Fargate Workloadmeta collector.
+package ecsfargate
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/testutil"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v3or4"
+)
+
+func TestPullWithTaskCollectionEnabledWithV4Parser(t *testing.T) {
+	// Start a dummy Http server to simulate ECS Fargate metadata v4 endpoint
+	dummyECS, err := testutil.NewDummyECS(
+		testutil.FileHandlerOption("/v4/1234-1/task", "./testdata/redis.json"),
+	)
+	require.Nil(t, err)
+	ts := dummyECS.Start()
+	defer ts.Close()
+
+	store := &fakeWorkloadmetaStore{}
+	// create an ECS Fargate collector with orchestratorECSCollectionEnabled enabled
+	collector := collector{
+		store:                 store,
+		taskCollectionEnabled: true,
+		metaV4:                v3or4.NewClient(fmt.Sprintf("%s/v4/1234-1", ts.URL), "v4"),
+	}
+	collector.taskCollectionParser = collector.parseTaskFromV4Endpoint
+
+	err = collector.Pull(context.Background())
+	require.Nil(t, err)
+	// one ECS task event and three container events should be notified
+	require.Len(t, store.notifiedEvents, 4)
+
+	count := 0
+	for _, event := range store.notifiedEvents {
+		require.Equal(t, workloadmeta.EventTypeSet, event.Type)
+		require.Equal(t, workloadmeta.SourceRuntime, event.Source)
+		switch entity := event.Entity.(type) {
+		case *workloadmeta.ECSTask:
+			require.Equal(t, 123457279990, entity.AWSAccountID)
+			require.Equal(t, "us-east-1", entity.Region)
+			require.Equal(t, "ecs-cluster", entity.ClusterName)
+			require.Equal(t, "RUNNING", entity.DesiredStatus)
+			require.Equal(t, "my-redis", entity.Family)
+			require.Equal(t, "1", entity.Version)
+			require.Equal(t, workloadmeta.ECSLaunchTypeFargate, entity.LaunchType)
+			count++
+		case *workloadmeta.Container:
+			require.Equal(t, "RUNNING", entity.KnownStatus)
+			require.Equal(t, "awslogs", entity.LogDriver)
+			require.Len(t, entity.Networks, 1)
+			require.Equal(t, "awsvpc", entity.Networks[0].NetworkMode)
```

Comment:
why didn't we assert these for the case of v2?

Reply:
https://github.com/DataDog/datadog-agent/pull/23253/commits/49535597cf93ec0442357589b38303da0fb1798c